fallback.go

```diff
@@ -54,10 +65,13 @@ func (f Fallback) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	}
 
 	if u, ok := f.rules[rules_index]; ok {
-		p := f.proxy.New(f.trace, u)
+		p := f.proxy.New(f.trace, u.proxyUpstream)
 		if p == nil {
 			return dns.RcodeServerFailure, fmt.Errorf("cannot create fallback proxy")
 		}
+		if u.original && originalRequest != nil {
```

Comment:
If ORIGINAL is set for current rule then use previously copied original request.